test/smtcallback.js

```diff
@@ -0,0 +1,155 @@
+const tape = require('tape');
+const fs = require('fs');
+const path = require('path');
+const solc = require('../index.js');
+const smtchecker = require('../smtchecker.js');
+const smtsolver = require('../smtsolver.js');
+
+var pragmaSol = 'pragma solidity >=0.0;\n';
+var pragmaSMT = 'pragma experimental SMTChecker;\n';
+
+function collectErrors (solOutput) {
+  if (solOutput === undefined) {
+    return [];
+  }
+
+  var errors = [];
+  for (var i in solOutput.errors) {
+    var error = solOutput.errors[i];
+    if (error.message.includes('This is a pre-release compiler version')) {
+      continue;
+    }
+    errors.push(error.message);
+  }
+  return errors;
+}
+
+function expectErrors (errors, expectations) {
+  if (errors.length !== expectations.length) {
+    return false;
+  }
+
+  for (var i in errors) {
+    if (!errors[i].includes(expectations[i])) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+tape('SMTCheckerCallback', function (t) {
+  t.test('Interface via callback', function (st) {
+    var satCallback = function (query) {
+      return { contents: 'sat\n' };
+    };
+    var unsatCallback = function (query) {
+      return { contents: 'unsat\n' };
+    };
+    var errorCallback = function (query) {
+      return { error: 'Fake SMT solver error.' };
+    };
+
+    var input = { 'a': { content: pragmaSol + pragmaSMT + 'contract C { function f(uint x) public pure { assert(x > 0); } }' } };
+    var inputJSON = JSON.stringify({
+      language: 'Solidity',
+      sources: input
+    });
+    var tests = [
+      { cb: satCallback, expectations: ['Assertion violation happens here'] },
+      { cb: unsatCallback, expectations: [] },
+      { cb: errorCallback, expectations: ['BMC analysis was not possible'] }
+    ];
+    for (var i in tests) {
+      var test = tests[i];
+      var output = JSON.parse(solc.compileStandardWrapper(
```

Comment:
Use `solc.compile`. `compileStandard` and `compileStandardWrapper` are deprecated aliases (should have removed them in 0.5.2, but will do in 0.6)